src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryParameterValueResolver.php

```diff
@@ -59,20 +63,25 @@ public function resolve(Request $request, ArgumentMetadata $argument): array
             'options' => $attribute->options,
         ];
 
-        if ('array' === $argument->getType() || $argument->isVariadic()) {
+        if ('array' === $type || $argument->isVariadic()) {
             $value = (array) $value;
             $options['flags'] |= \FILTER_REQUIRE_ARRAY;
         } else {
             $options['flags'] |= \FILTER_REQUIRE_SCALAR;
         }
 
-        $filter = match ($argument->getType()) {
+        $enumType = null;
+        $filter = match ($type) {
             'array' => \FILTER_DEFAULT,
             'string' => \FILTER_DEFAULT,
             'int' => \FILTER_VALIDATE_INT,
             'float' => \FILTER_VALIDATE_FLOAT,
             'bool' => \FILTER_VALIDATE_BOOL,
-            default => throw new \LogicException(sprintf('#[MapQueryParameter] cannot be used on controller argument "%s$%s" of type "%s"; one of array, string, int, float or bool should be used.', $argument->isVariadic() ? '...' : '', $argument->getName(), $argument->getType() ?? 'mixed'))
+            default => match ($enumType = is_subclass_of($type, \BackedEnum::class) ? (new \ReflectionEnum($type))->getBackingType()->getName() : null) {
```

Comment:
Now I see that `ReflectionType` is abstract and you get `ReflectionNamedType` always. I was convinced that you get `ReflectionNamedType` since 8.2.0 only, that is why I've used `::_toString`.